anvil/scripts/anvil_train.py

```diff
@@ -10,7 +10,7 @@
 
 log = logging.getLogger(__name__)
 
-PROVIDERS = ["anvil.train", "anvil.checkpoint",]
+PROVIDERS = ["anvil.train", "anvil.checkpoint", "anvil.models", "anvil.plot"]
```

Comment:
models doesn't look necessary here and removing it doesn't do anything

Reply:
it's needed for plot_distributions to work but nothing else

Reply:
But I can remove it and the benchmark still runs and as far as I can tell there are no 'actions' in it asides from the things which we set as explicit nodes

Reply:
ah sorry, my mistake - it's also needed for training O(2)/O(3) since `stereographic_projection` action takes `real_nvp` as a parameter.

Reply:
In future I would want to replace this dependency on `real_nvp` with a dependency on something like `inner_flow` which could be an explicit node defined in config.py, since the projection map can wrap around any flow on a compact domain (e.g. splines). I could do this now.

Reply:
I'm not sure how this fits together with your suggestion of specifying the entire model sequentially in the runcard. Maybe it makes the case for doing that stronger.

Reply:
> I'm not sure how this fits together with your suggestion of specifying the entire model sequentially in the runcard. Maybe it makes the case for doing that stronger.

This, I was just about to say that I think this PR has already highlighted the need for that more flexible system. Perhaps we can call and chat about this?

Reply:
Ah yeah sure, a chat would be good. Although, the comment you've quoted, and the one before, I've deleted because I got confused. I said the projection could wrap around any old flow on a compact space, but it's the other way round - it wraps around flows on R^N. I think Real NVP is likely to be the only thing we use on R^N in the near future, so we can probably get away without this extra flexibility for now.